File: greedy.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

void pyramid(int input_pyramid_height);
int main(void)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    float pay_in1;
    int coins_used = 0;
    int total_coins = 0;
    do
    {
         printf("O hai! How much change is owed?\n");
         pay_in1 = get_float();
    }
    while(pay_in1 < 0);
    int pay_in = roundf(pay_in1*100);
    
    if(pay_in >= quarter){ //
        coins_used = (pay_in / quarter);
        total_coins = coins_used + total_coins;
        pay_in = pay_in - (coins_used * quarter);
    }
    if(pay_in >= dime){ // 
        coins_used = (pay_in / dime);
        total_coins = coins_used + total_coins;
        pay_in = pay_in - (coins_used * dime); // 0.16 - 
    } 
    if(pay_in >= nickel){
        coins_used = (pay_in / nickel);
        total_coins = coins_used + total_coins;
        pay_in = pay_in - (coins_used * nickel);
    } 
    if(pay_in >= penny){
        coins_used = (pay_in / penny);
        total_coins = coins_used + total_coins;
        pay_in = pay_in - (coins_used * penny);
    }    
    int coins_output = total_coins;
    printf("%i\n",coins_output);
}

File: greedy.py
import cs50
import math


QUARTER = 0.25
DIME = 0.10
NICKEL = 0.05
PENNY = 0.01
TOTAL_COINS = 0
COINS_USED = 0
PAY_IN = 0

def get_positive_float():
    while True:
        n = cs50.get_float()
        if n > 0:
            break
    return n

def left_over_calc(coinType):

    global PAY_IN
    global TOTAL_COINS
    global COINS_USED

    COINS_USED =  math.floor( PAY_IN / coinType)
    TOTAL_COINS = COINS_USED +  TOTAL_COINS
    PAY_IN = round(PAY_IN - (COINS_USED *  coinType),2)

def main():
    print('O hai! How much change is owed?')
    global PAY_IN
    PAY_IN = get_positive_float()

    if(PAY_IN >= QUARTER):
        left_over_calc(QUARTER)
    if(PAY_IN >= DIME):
        left_over_calc(DIME)
    if(PAY_IN >= NICKEL):
        left_over_calc(NICKEL)
    if(PAY_IN >= PENNY):
        left_over_calc(PENNY)
    print(TOTAL_COINS)

if __name__ == "__main__":
    main()
